src/editor/C4EditCursor.cpp

```diff
@@ -880,7 +880,9 @@ void C4EditCursor::Draw(C4TargetFacet &cgo)
 	if (Mode == C4CNS_ModeCreateObject && has_mouse_hover && creator_def)
 	{
 		C4TargetFacet cgo_creator;
-		cgo_creator.Set(cgo.Surface, X + cgo.X - cgo.TargetX, Y + cgo.Y - cgo.TargetY,
+		cgo_creator.Set(cgo.Surface,
+			X + cgo.X - cgo.TargetX + creator_def->Shape.x + creator_def->Shape.Wdt / 2,
+			Y + cgo.Y - cgo.TargetY + creator_def->Shape.y + creator_def->Shape.Hgt / 2,
 			creator_def->Shape.Wdt, creator_def->Shape.Hgt, 0, 0, cgo.Zoom, 0, 0);
```

Comment:
Could you explain what this does?

Also, just guessing now, I think that it works for objects where the offset is symmetrical i.e. Offset = -20, -5, Width = 40, Height = 10. What about an object with Offset = -10, -2, Width = 40, Height = 10?

Reply:
This is to fix object preview where Offset is not (`-Width / 2`, `-Height / 2`)

I think the current code assumed that Offset is always `-Wdt / 2` (or Height, resp.)
So I basically add the real offset (which is negative usually) and compensate by adding `Wdt / 2`.
As far as I can tell, this looks right for all existing objects.

Reply:
Add an explaining comment then, please